src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -240,15 +240,26 @@ void BLEManagerImpl::DriveBLEState()
         {
             mFlags.Clear(Flags::kAdvertisingRefreshNeeded);
             err = StartAdvertising();
-            SuccessOrExit(err);
+            if (err != CHIP_NO_ERROR)
+            {
+                // Return prematurely but keep the CHIPoBLE service mode enabled to allow advertising retries
+                mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Enabled;
```

Comment:
This is the case when an error occurs, right? So why do we set mServiceMode to Enabled?

Reply:
This is to keep the service mode globally enabled to be able to restart the advertising after it fails. This is how the DriveBLEState logic works. But we do clear the kChipoBleGattServiceRegister and unregister the service in case of a failure.